docs/adr/adr-11882-publication-of-providers-package.md

```diff
@@ -0,0 +1,138 @@
+
+# ADR-11882: Process for publishing the  az  provider types (alpha)
+
+* **Date:** 2023-09-07
+* **Status:**  Accepted (Bicep Discussions 09/27/2023)
+
+## Context
+
+This document aims to formally describe an initial process for publishing the az resource types provider packages
+as OCI artifacts in the Bicep registry so they can be dynamically loaded as described in the proposal
+https://github.com/Azure/bicep/issues/10662 (previously accepted by the Bicep maintainers) for the alpha milestone.
+
+### Current State
+
+The current state has two github workflows:
+
+1. **build.yml** - Triggered by PRs and push to `main`. Builds the `generator` and its dependencies and publishes a NuGet
+package with the Azure resource type definitions.
+<!-- //TODO(asilverman): Describe the versioning scheme for the NuGet package -->
+
+1. **update-types.yml** - Triggered by a cron schedule (once a week) or manually. Checks out a snapshot of the `main`
+branch from the `azure/azure-rest-api-specs` repo, runs the generation script on the snapshot and commits the result to the
+[`/generated`](https://github.com/Azure/bicep-types-az/tree/autogenerate/generated) folder in the `autogenerate` branch or  
+`azure/bicep-types-az/tree/autogenerate-<git-commit-ish>`  branches depending on the trigger (a '`<git-commit-ish>`' is
+a truncated commit sha256 digest)
+
+When a new version of the Bicep is close to being released, the person cutting a release performs the following steps:
+
+1. Chose to either manually merge the autogenerate branch into `main` or manually trigger the **update-types.yml**
+workflow to update the `/generated` folder. This is typically reviewed, but not in great detail.
+(hard to review 600+ automated file changes!)
+
+1. This will update the definitions stored in `/generated` in the `main`
+branch and trigger the **build.yml** workflow which in turn will publish a NuGet package with a bumped version.
+
+1. Verify (manually) that a new NuGet package was published
+
+1. Update the NuGet package reference in the  `Bicep.Core.csproj` file in the `azure/bicep`  repo to the new package
+version
+
+1. Proceed with the release process for  `azure/bicep`
+
+The current process allow us to manually inspect and catch any odd regression before impacting any customers
+(e.g. the time where we accidentally deleted all of the types for `Microsoft.Compute`)
```

Comment:
Should we be able to have automated tests that verify nothing has been deleted?

Reply:
yes, but these kids of tests are probably in the scope of the `generator` script . Would you mind creating a separate task for that?

Reply:
np